### DISCUSSION TITLE
api call "/stats/code_frequency" returns empty object until you open "/graphs/code-frequency" in web browser

### DISCUSSION THREAD

**Discussion by freeart** [community/community#59744] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I work in a corporation, and we have a lot of internal and private projects. Each of them returns {} for https://api.github.com/repos/ORG/REPO/stats/code_frequency until I navigate to https://github.com/ORG/REPO/graphs/code-frequency in a web browser. After that, the API call returns an array of stats.

**Comment by tomch3ng** [comment]
I encountered the same issue, but upon further research, I saw this note in the [API documentation](https://docs.github.com/en/rest/metrics/statistics?apiVersion=2022-11-28#best-practices-for-caching): 
> Computing repository statistics is an expensive operation, so we try to return cached data whenever possible. If the data hasn't been cached when you query a repository's statistics, you'll receive a 202 response; a background job is also fired to start compiling these statistics. You should allow the job a short time to complete, and then submit the request again.

I confirmed that waiting a few minutes and then resubmitting the request does work. This also fits with the workaround you discovered, that pulling up the code frequency graph via your browser will cause Github to fetch and cache the statistics.

**Reply by StanBright** [reply]
Thanks @tomch3ng. I've been pulling my hair for a week or two resolving an issue related to this one. The API client (Oktokit.rb) would simply return `nil`, randomly, in those cases.

**Comment by xs2bharat** [comment]
for me even after browsing to "/graphs/code-frequency", api still returns empty object
